### PR TITLE
libmpdclient/mpc/mpd: update and build with meson

### PR DESCRIPTION
**Pull request by dhewg**
Maintainer: @thess 
Compile tested: cubieboard2
Run tested: cubieboard2

Depends on #10043 #10113

### CODE REVIEW

**Comment by neheb** [comment]
Needs a rebase.

Do you know why boost-container is misted as a dependency?

**Comment by dhewg** [comment]
Rebased, and it looks like boost-container is a leftover? Removed it, works here, but circleci keeps on running in circles

**Comment by neheb** [comment]
CircleCI fails on compiling boost. Too little RAM.

**Comment by dhewg** [comment]
But now with boost-container gone, why does it compile boost crap at all? It's now just the header files that's required, right?

**Comment by neheb** [comment]
mpd uses boost. both the headers and libraries are needed. the extra libraries provided by boost-container are not.

**Comment by dhewg** [comment]
I didn't look too closely, but I think it would be sufficient for mpd to have the boost headers. It's not linking against any boost library. CI wouldn't need to compile anything boost, but maybe that's just the way the boost package is designed. It's just a shame that I cannot throw my meson changes at the CI because of boost...

**Comment by neheb** [comment]
Please rebase against current master. I have this in my local branch. I think it builds.

**Comment by dhewg** [comment]
rebased

**Comment by GNUuser** [comment]
Hello @dhewg 
thanks for updating MPD, actually the version 0.20.23 has ugly bugs like stutering, eat-of-bit memory-leaks and  more. Updating to top version could be nice... i am pending on this PR.

**Comment by dhewg** [comment]
update:
* update to mpd v0.21.16, which includes the build fix included as patch earlier

File: sound/mpc/Makefile [hunk 1]
```diff
@@ -6,21 +6,24 @@
...
-PKG_HASH:=65fc5b0a8430efe9acbe6e261127960682764b20ab994676371bdc797d867fce
+PKG_HASH:=7961d95b7ce019996beab281cf957e905667c989c53fffd13ade5e62fea331c7
 
 PKG_LICENSE:=GPL-2.0-or-later
 PKG_LICENSE_FILES:=COPYING
 PKG_MAINTAINER:=Ted Hess <thess@kitschensync.net>
 
-PKG_FIXUP:=autoreconf
+PKG_BUILD_DEPENDS:=meson/host
 PKG_INSTALL:=1
```
Comment (neheb) [inline sound/mpc/Makefile]: PKG_BUILD_PARALLEL could be good here. I don't know if Meson has problems with it.

File: libs/libmpdclient/Makefile [hunk 2]
```diff
@@ -17,10 +17,11 @@ PKG_MAINTAINER:=Ted Hess <thess@kitschensync.net>
 PKG_LICENSE:=BSD-3-Clause
 PKG_LICENSE_FILES:=COPYING
 
-PKG_FIXUP:=autoreconf
+PKG_BUILD_DEPENDS:=meson/host
 PKG_INSTALL:=1
```
Comment (neheb) [inline libs/libmpdclient/Makefile]: PKG_BUILD_PARALLEL could be good here. No idea if meson has problems with it.

File: sound/mpd/Makefile [hunk 3]
```diff
@@ -6,19 +6,17 @@
...
-PKG_SOURCE_URL:=http://www.musicpd.org/download/mpd/0.20/
-PKG_HASH:=503e5f9f237290f568ff7956ab2f9aed563594bf749f19b8fe994fb21434afea
+PKG_SOURCE_URL:=http://www.musicpd.org/download/mpd/0.21/
+PKG_HASH:=30cf1bddf7d7388487276745ad3515f134e07f0c57f9f97cb2b5d3befd4a4d92
 
 PKG_MAINTAINER:=Ted Hess <thess@kitschensync.net>
 PKG_LICENSE:=GPL-2.0-or-later
 PKG_LICENSE_FILES:=COPYING
 
-PKG_BUILD_PARALLEL:=1
```
Comment (neheb) [inline sound/mpd/Makefile]: Does meson have problems with this?

File: sound/mpd/Makefile [hunk 4]
```diff
@@ -100,90 +99,110 @@ endef
...
+	-Daudiofile=disabled \
+	-Dfaad=disabled \
+	-Dflac=enabled \
+	-Dfluidsynth=disabled \
+	-Dgme=disabled \
+	-Dmad=disabled \
+	-Dmikmod=disabled \
+	-Dmodplug=disabled \
+	-Dmpcdec=disabled \
+	-Dopus=disabled \
```
Comment (neheb) [inline sound/mpd/Makefile]: Interesting that opus is disabled... I wonder if this is from back then libopus was not available in the tree.

Comment (neheb) [reply sound/mpd/Makefile]: Or maybe libffmpeg decodes it.

Comment (diizzyy) [reply sound/mpd/Makefile]: Unless you need to encode just let ffmpeg handle everything

File: sound/mpc/Makefile (hunk 1 above)
Comment (dhewg) [reply sound/mpc/Makefile]: It does builds via ninja in parallel, just doesn't require PKG_BUILD_PARALLEL, that's a gnu make setting that's not required with meson.

File: sound/mpd/Makefile (hunk 4 above)
Comment (dhewg) [reply sound/mpd/Makefile]: I too noticed that opus wasn't enabled and figured that's from a time where that package wasn't available.
My idea was to extend this package further after this MR to allow it, this just keeps the exact same feature sets with meson.

**Pull request merged**